src/mpl/src/thread/mpl_thread_posix.c

```diff
@@ -53,9 +57,46 @@ void MPL_thread_create(MPL_thread_func_t func, void *data, MPL_thread_id_t * idp
         pthread_attr_init(&attr);
         pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED);
 
+#if MPL_THREAD_PACKAGE_NAME == MPL_THREAD_PACKAGE_UTI
+		uti_attr_t* uti_attr = NULL;
+		uti_value_t val;
+		err = uti_attr_create(uti_attr);
+		if(err) {
+			goto uti_exit;
+		}
+
+		/* Give a hint that it's beneficial to put the thread
+		   on the same NUMA-node as the creator */
+		val.type = UTI_TYPE_BOOL;
+		val.flag = true;
+		err = uti_attr_put(uti_attr, "same_numa_node", &val);
+		if(err) {
+			goto uti_exit;
+		}
+
+		/* Give a hint that the thread repeatedly monitors a device. */
+		val.type = UTI_TYPE_ENUM;
+		val.behavior = UTI_BEHAVIOR_POLLING;
+		err = uti_attr_put(uti_attr, "behavior", &val);
+		if(err) {
+			goto uti_exit;
+		}
+
+        err = uti_pthread_create(idp, &attr, MPLI_thread_start, thread_info, uti_attr);
+		if(err) {
+			goto uti_exit;
+		}
+
+		err = uti_attr_destroy(uti_attr);
```

Comment:
Should we call this in the error cases as well? It's currently being skipped over if `uti_attr_put` or `uti_pthread_create` fail.